### DISCUSSION TITLE
Can gnpy calculate the interaction of services？

### DISCUSSION THREAD

**Discussion by ZYIAO** [Telecominfraproject/oopt-gnpy#440] [Q&A] [answered]
hello,
Thank you for this great and useful tool.
I‘m learning and using GNPy to calculate SNR in optical networks. In network level computing, there are many services, can gnpy calculate the interaction of services？I give an example here,

![example](https://user-images.githubusercontent.com/103397154/162686722-ded62049-1350-4f69-8cb3-8addb9c98790.png)

As described in the picture, Service 1 starts from node A to node B, then we add Service 2 in. After that, Service 1 and Service 2 will transmit on different channels from node B to node C together. I think, on the link B->C, the service 1 and 2 will affect each other, including XPM and SRS, etc.
I used the path_requests_run function, it seems that there is no such computed results, nor in LYNX web simulation.
The results of Service 1 when 1 and 2 are calculated together are the same as that of only calculating Service 1. This is not true in the actual network. 
So, can gnpy calculate the interaction of services？Or how should I modify it?

Another question, how to describe the empty channels?
Thanks !

**Comment by jktjkt** [comment, marked as answer]
Hi ZYIAO, this is indeed supported in GNPy. In fact, by default GNPy will assume "full channel load" (as defined via the `SpectrumInfo` statement), even when some channels are not populated in the real network. In your example, this means that there will be not just Service 1 and Service 2, but also services 3 to 76. That's very useful for end-of-life simulations, worst-case network dimensioning, etc.

We have patches under review which will make it easier to define an arbitrary spectrum, mixed rates of channels, etc.

**Comment by oolongkiller** [comment]
Hi ZYIAO, have you successfully implemented the situation described above? I currently also want to implement the service shown by you, but I still don't understand how to utilize Gnpy at the moment. 

Could you give me some guidance?
